include/libtorrent/ip_notifier.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+
+Copyright (c) 2016, Steven Siloti
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+* Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in
+the documentation and/or other materials provided with the distribution.
+* Neither the name of the author nor the names of its
+contributors may be used to endorse or promote products derived
+from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef TORRENT_IP_NOTIFIER_HPP_INCLUDED
+#define TORRENT_IP_NOTIFIER_HPP_INCLUDED
+
+#include <functional>
+#include <array>
+
+#include "libtorrent/error_code.hpp"
+#include "libtorrent/address.hpp"
+#include "libtorrent/io_service.hpp"
+
+#if defined TORRENT_BUILD_SIMULATOR
+#elif TORRENT_USE_NETLINK
+#include "libtorrent/netlink.hpp"
+#elif defined TORRENT_WINDOWS
+#include <boost/asio/windows/object_handle.hpp>
+#endif
+
+namespace libtorrent
+{
+	struct ip_change_notifier : boost::noncopyable
+	{
+		explicit ip_change_notifier(io_service& ios);
+		~ip_change_notifier();
+
+		// cb will be invoked  when a change is detected in the
+		// system's IP addresses
+		void async_wait(std::function<void(error_code const&)> cb);
+		void cancel();
+
+	private:
+		void on_notify(error_code const& error
+			, std::size_t bytes_transferred
+			, std::function<void(error_code const&)> cb);
+
+#if defined TORRENT_BUILD_SIMULATOR
+		// TODO simulator support
+#elif TORRENT_USE_NETLINK
+		netlink::socket m_socket;
+		std::array<char, 4096> m_buf;
+#elif defined TORRENT_WINDOWS
+		OVERLAPPED m_ovl;
```

Comment:
this member doesn't seem to be initialized or cleared completely. afaict, only the event is set on it. Can you add `` = {}`` here to zero-initialize it?